rero_ils/modules/loans/api.py

```diff
@@ -761,7 +761,11 @@ def get_notification_candidates(self, trigger):
 
         candidates = []
         item = Item.get_record_by_pid(self.item_pid)
-        has_request = item.number_of_requests() > 0
+        # Get the number of requests on the related item and exclude myself
+        # from the result list.
+        requests = item.get_requests(pids=True)
+        requests = [loan_pid for loan_pid in requests if loan_pid != self.pid]
```

Comment:
You can also use filer if you want;
```python
requests = filter(lambda x: x != self.pid, requests)
```

Reply:
list comprehension seems more efficient than filter + lambda. More : filter return a genetator and you need to cast it to a list again. I read somewhere than Python guru (Guido) should have liked to remove `filter` from python3